models.py

```diff
@@ -100,17 +101,29 @@ def annotations_by_books_and_prints(self, group_amount=50):
 
         return (books,prints)
 
+    def format_trp_id(self, trp_id):
+        return '%04d' % int(trp_id)
 
     def _get_trp_id(self):
-        last_bio = Biography.objects.order_by('-trp_id')[0]
+        ordered_bios = Biography.objects.order_by('-trp_id')
+        index = 0
+        last_bio = ordered_bios[index]
+        while last_bio.trp_id != self.format_trp_id(last_bio.trp_id):
+            last_bio = ordered_bios[index+1]
         last_trp_id = int(last_bio.trp_id)
         new_trp_id = last_trp_id + 1
         return '%04d' % new_trp_id
 
     def save(self, *args, **kwargs):
         if not self.trp_id:
             self.trp_id = self._get_trp_id()
-        super(Biography, self).save(*args, **kwargs)
+        else:
+            self.trp_id = self.format_trp_id(self.trp_id)        
+        self.roles = self.roles.replace(",",";"); 
+        try:
+            super(Biography, self).save(*args, **kwargs)
+        except IntegrityError as e:
+            print e.message
```

Comment:
It looks to me like if there's an IntegrityError, we would just print out the message and silently fail. Shouldn't we re-raise the exception, or raise a new one, so that it's clear that the Biography didn't get saved?